include/GafferScene/Shader.h

```diff
@@ -121,6 +121,9 @@ class Shader : public Gaffer::DependencyNode
 		/// \deprecated
 		IECore::ConstObjectVectorPtr state() const;
 
+		virtual void attributesHash( IECore::MurmurHash &h, const Gaffer::Plug *output ) const;
```

Comment:
Thanks for detailed explanation - I'm not sure if I fully understand but I'll refer back to these notes while reading the code.